homeassistant/components/jewish_calendar/config_flow.py

```diff
@@ -0,0 +1,158 @@
+"""Config flow for Jewish calendar integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+import zoneinfo
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_ELEVATION,
+    CONF_LANGUAGE,
+    CONF_LOCATION,
+    CONF_NAME,
+    CONF_TIME_ZONE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    LocationSelector,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_CANDLE_LIGHT_MINUTES,
+    CONF_DIASPORA,
+    CONF_HAVDALAH_OFFSET_MINUTES,
+    DEFAULT_CANDLE_LIGHT,
+    DEFAULT_DIASPORA,
+    DEFAULT_HAVDALAH_OFFSET_MINUTES,
+    DEFAULT_LANGUAGE,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+
+LANGUAGE = [
+    SelectOptionDict(value="hebrew", label="Hebrew"),
+    SelectOptionDict(value="english", label="English"),
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, str]:
+    """Validate the user input."""
+    return {"title": data[CONF_NAME]}
+
+
+class JewishCalendarConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Jewish calendar."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return JewishCalendarOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if all(
+                    _value == entry.as_dict().get(_key)
+                    for _key, _value in user_input.items()
+                    if _key != CONF_NAME
+                ):
+                    return self.async_abort(reason="already_configured")
+
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            try:
+                info = await validate_input(self.hass, user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
```

Comment:
You should make the schema a constant at the top of the file.
It will make the code easier to read.

Reply:
This is resolved